drivers/serial/uart_kb1200.c

```diff
@@ -0,0 +1,689 @@
+/*
+ * Copyright (c) 2021, Thomas Stranger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * This	is not a real serial driver. It	is used	to instantiate struct
+ * devices for the "vnd,serial"	devicetree compatible used in test code.
+ */
+
+#define	DT_DRV_COMPAT ene_kb1200_uart
+
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/ring_buffer.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <soc.h>
+
+struct kb1200_uart_config {
+	uintptr_t base_addr;
+	uint32_t port_num;
+	const struct pinctrl_dev_config *pcfg;
+};
+
+struct kb1200_uart_data	{
+	uart_irq_callback_user_data_t callback;
+	void *callback_data;
+	uint8_t	PendingFlag_data;
+};
+
+/* UART	Interface (commmon API)	*/
+/**********************************************************************************
+ * Check whether an error was detected.
+ *
+ * Function: int uart_err_check	(const struct device *dev)
+ *
+ * Return: 0	    If no error	was detected.
+ *	   err	    Error flags	as defined in uart_rx_stop_reason
+ *	   -ENOSYS  If not implemented
+ **********************************************************************************/
+static int kb1200_uart_err_check(const struct device *dev)
+{
+	const struct kb1200_uart_config *config = dev->config;
+	SER_T* ser = (SER_T *)config->base_addr;
+	printk("kb1200_uart_err_check\n");
+	int err=0;
+	if(ser->SERSTS & SER_RxOverRun)
+		err |= UART_ERROR_OVERRUN;
+	if(ser->SERSTS & SER_ParityError)
+		err |= UART_ERROR_PARITY;
+	if(ser->SERSTS & SER_FramingError)
+		err |= UART_ERROR_FRAMING;
```

Comment:
Coding styling says that brackets are needed even for one line if block. Could you go through all changes in this PR to make sure brackets are added?